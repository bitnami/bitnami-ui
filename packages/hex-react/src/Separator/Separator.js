import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classnames';

const Separator = props => {
  const { size, className, margin, width, ...attributes } = props;

  const css = cs(className, {
    [`separator-${size}`]: size != null,
    [`margin-v-${margin}`]: margin != null,
    [`separator-${width}`]: width != null,
  });

  return React.createElement('hr', { className: css, ...attributes });
};

Separator.propTypes = {
  /**
   * Modify the height of the separator
   */
  size: PropTypes.oneOf(['small', 'big', 'bigger', 'enormous']),
  /**
   * Modify the vertical margin for the separator
   */
  margin: PropTypes.oneOf(['small', 'big', 'bigger', 'enormous', 'giant', 'huge']),
  /**
   * Modify the width (percentage) of the separator
   */
  width: PropTypes.oneOf(['15', '30', '50', '70']),
  /**
   * Other CSS classes to add to the component
   */
  className: PropTypes.string,
};

Separator.defaultProps = {
  size: undefined,
  margin: undefined,
  width: undefined,
  className: '',
};

export default Separator;
