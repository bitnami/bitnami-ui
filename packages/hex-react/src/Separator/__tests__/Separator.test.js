import React from 'react';
import { shallow } from 'enzyme';

import Separator from '../Separator';

describe('Separator', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Separator />);
    expect(wrapper).toMatchSnapshot();
  });

  it('adds a custom CSS class', () => {
    const css = 'test';
    const wrapper = shallow(<Separator className={css} />);
    expect(wrapper.hasClass(css)).toBe(true);
  });

  it('sets the correct size', () => {
    const wrapper = shallow(<Separator size="small" />);
    expect(wrapper.hasClass('separator-small')).toBe(true);
  });

  it('sets the correct width', () => {
    const wrapper = shallow(<Separator width="50" />);
    expect(wrapper.hasClass('separator-50')).toBe(true);
  });

  it('sets the correct margin', () => {
    const wrapper = shallow(<Separator margin="huge" />);
    expect(wrapper.hasClass('margin-v-huge')).toBe(true);
  });
});
