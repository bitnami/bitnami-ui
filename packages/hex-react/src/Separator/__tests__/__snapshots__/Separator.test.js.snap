// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders correctly 1`] = `
<hr
  className=""
/>
`;
